File: object/object/src/Lec02/condition/refund/TimeCondition.java
package Lec02.condition.refund;


import Lec02.Screening;

import java.time.LocalDateTime;

public class TimeCondition implements RefundCondition {

    private LocalDateTime currentTime;

    @Override
    public boolean isSatisfiedBy(Screening screening) {
        return currentTime.compareTo(screening.getStartTime()) <= 0;
    }
}

File: object_Movie/src/main/java_to_kotlin/condition/refund/TimeCondition.kt
package java_to_kotlin.condition.refund

import java_to_kotlin.Refund
import java.time.LocalDateTime

class TimeCondition(private val currentTime: LocalDateTime) : RefundCondition {
    override fun isSatisfiedBy(refund: Refund): Boolean {
        return currentTime.compareTo(refund.movieStartTime) <= 0
    }
}
